docs/commands.md

```diff
@@ -131,7 +131,27 @@ A site-wide commandfile should have tests that run with each (major) version of
 
 Commandfiles that are not part of any Drupal site are called 'global' commandfiles.
 
-### Commands discovered by configuration
+### Auto-discovered commands
+
+Such commands are auto-discovered by their class PSR4 namespace and class/file name suffix. Drush will auto-discover commands if:
+
+* The commands class is PSR4 auto-loadable.
+* The commands class namespace, relative to base namespace, is `Drush\Commands`. For instance, if a Drush command provider third party library maps this PSR4 autoload entry:
+  ```json
+  "autoload": {
+    "psr-4": {
+      "My\\Custom\\Library\\": "src"
+    }
+  }
+  ```
+  then the Drush global commands class namespace should be `My\Custom\Library\Drush\Commands` and the class file should be located under the `src/Drush/Commands` directory.
+* The class and file name ends with `*DrushCommands`, e.g. `FooDrushCommands`.
+
+Unlike commands defined in Drupal modules, global commands don't benefit from dependency injection. However, it's still possible to interact with Drupal container services by using the `\Drupal::services()` static method. This can be achieved by defining the bootstrap level in the command attributes (see https://www.drush.org/latest/bootstrap) for details. A global command needing access to Drupal services should use the `DrupalBootLevels::FULL` bootstrap level.
```

Comment:
- The method is service(), not services() (not plural)
- Since 'bootstrap' is a link from one .md file to another .md file, you should use a relative path instead of an url.